src/main/java/com/apigee/edge/config/mavenplugin/CustomRoleMojo.java

```diff
@@ -0,0 +1,352 @@
+		} catch (RuntimeException e) {
+			throw e;
+		}
+	}
+
+	/**
+	 * Entry point for the mojo.
+	 */
+	public void execute() throws MojoExecutionException, MojoFailureException {
+
+		if (super.isSkip()) {
+			getLog().info("Skipping");
+			return;
+		}
+
+		Logger logger = LoggerFactory.getLogger(CustomRoleMojo.class);
+
+		try {
+
+			init();
+
+			if (buildOption == OPTIONS.none) {
+				logger.info("Skipping Custom Roles (default action)");
+				return;
+			}
+
+			if (serverProfile.getEnvironment() == null) {
+				throw new MojoExecutionException("Apigee environment not found in profile");
+			}
+			List customRoles = getOrgConfig(logger, userRole);
+			if (customRoles == null || customRoles.size() == 0) {
+				logger.info("No Custom Role config found.");
+				return;
+			}
+
+			doUpdate(customRoles);
+
+		} catch (MojoFailureException e) {
+			throw e;
+		} catch (RuntimeException e) {
+			throw e;
+		}
+	}
+
+	/***************************************************************************
+	 * REST call wrappers
+	 **/
+	public static String createCustomRole(ServerProfile profile, String customRole) throws IOException {
+
+		Gson gson = new Gson();
+		CustomRole customRoleObject = gson.fromJson(customRole, CustomRole.class);
+		String namePayload = customRoleConversion(customRoleObject.name);
+
+		HttpResponse response = RestUtil.createOrgConfig(profile, userRole, namePayload);
+		try {
+
+			logger.info("Response " + response.getContentType() + "\n" + response.parseAsString());
+			if (response.isSuccessStatusCode()) {
+				logger.info("Create Success for role.");
+				addResourcePermissionsToRole(profile, customRoleObject);
+			}
+
+		} catch (HttpResponseException e) {
+			logger.error("Custom Role create error " + e.getMessage());
+			throw new IOException(e.getMessage());
+		}
+
+		return "";
+	}
+
+	/**
+	 * This method Assign Multiple Resource Permissions to the user role .
+	 * 
+	 * @param profile
+	 * 
+	 * @param customRoleObject user role Object
+	 */
+	private static void addResourcePermissionsToRole(ServerProfile profile, CustomRole customRoleObject)
+			throws IOException {
+
+		String permissionsPayload = customRolePermissionConversion(customRoleObject.resourcepermissions);
+		HttpResponse response = RestUtil.createOrgConfig(profile,
+				userRole + "/" + customRoleObject.name + "/resourcepermissions", permissionsPayload);
+		try {
+
+			logger.info("Response " + response.getContentType() + "\n" + response.parseAsString());
+			if (response.isSuccessStatusCode()) {
+				logger.info("Create Success for role permissions.");
+			}
+
+		} catch (HttpResponseException e) {
+			logger.error("Custom Role create error " + e.getMessage());
+			throw new IOException(e.getMessage());
+		}
+	}
+
+	/**
+	 * This method is used to update the user role Permissions.
+	 *
+	 * @param profile
+	 *
+	 *
+	 * @param customRole
+	 */
+	public static String updateCustomRole(ServerProfile profile, String customRole) throws IOException {
```

Comment:
What happens when I create a custom role with 3 resource permissions and then update the json file to delete one resource. Now I run update ? What will it do ? Do you think it will remove the excluded permission from the role ? Was that tested ?

Reply:
It will update the  resource permissions to custom role and if we delete the resource permission from edge.json file then the resource permissions will not get deleted.
Running update command will only add new resource permissions to custom role. It will not remove the excluded permission from the role.
If we use the sync command then it will delete the excluded permission from the role. 
This behavior is in line with the behavior of the api(https://apidocs.apigee.com/management/apis/post/organizations/%7Borg_name%7D/userroles/%7Brole_name%7D/resourcepermissions)

Reply:
ok.. lets add that to the README